scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -0,0 +1,508 @@
+                  subdiagnosis,
+                  "ARMS" = "Alveolar rhabdomyosarcoma",
+                  "ERMS" = "Embryonal rhabdomyosarcoma"
+                ))
+
+# add back to sce 
+colData(integrated_sce) <- DataFrame(coldata_df, row.names = coldata_df$cell_id)
+```
+
+Now we can use that sample metadata when making our plots. 
+Lets start by looking at our different subdiagnoses.
+
+```{r}
+scater::plotReducedDim(integrated_sce,
+                       dimred = "fastmnn_UMAP",
+                       colour_by = "sample",
+                       point_size= 0.5,
+                       point_alpha = 0.2,
+                       other_fields = "diagnosis_group") +
+  facet_wrap(~diagnosis_group)
+```
+
+Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with each other rather than integrate all together. 
+
+We can also look at the distribution of cell types to see if these subdiagnoses have different or shared cell types.
+
+*Talk about looking across cell types and comparing cell types* 
+
+```{r}
+# facet the UMAP by sample to show the distribution of cell types across each subdiagnosis
+scater::plotReducedDim(integrated_sce,
+                       dimred = "fastmnn_UMAP",
+                       colour_by = "celltype_broad",
+                       point_size= 0.5, 
+                       point_alpha = 0.4,
+                       other_fields = "diagnosis_group") +
+  facet_wrap(~ diagnosis_group)
+```
+We can also look at the distribution of cell types across each sample.
+
+```{r}
+tumor_cells_df <- coldata_df %>%
+  dplyr::filter(stringr::str_detect(celltype_broad, "Tumor"))
+
+ggplot(tumor_cells_df, aes(x = sample, fill = celltype_broad)) + 
+    geom_bar(color = "black", size = 0.2) +
+    labs(
+      x = "Sample",
+      y = "Number of cells", 
+      fill = "Cell type"
+    ) +
+  theme_bw() +
+  theme(axis.text.x = element_text(angle = 90))+
+  facet_wrap(~diagnosis_group, scales = "free_x")
+```
+
+*Need to explain the difference between tumor/tumor_myoblast, etc*
+These plots show that ARMS and ERMS share a lot of the same cell types, but perhaps there are differences in the cells in each subdiagnoses.
+We also see that 6 of the libraries show a mixture of tumor cells belonging to the mesoderm, myoblast, and myocyte classes.
+3 libraries contain cells that are only classified as tumor or non-tumor, and tumor cells are not further classified and the remaining library appears to not have been assigned any cell types, as they are all labeled with `NA`.
+We will not be using these 3 libraries.
+
+```{r}
+# define libraries to keep
+library_ids <- c(
+  "SCPCL000479",
+  "SCPCL000480",
+  "SCPCL000481",
+  "SCPCL000484",
+  "SCPCL000488",
+  "SCPCL000491"
+)
+
+# subset sce to only contain samples with library IDs of interest 
+samples_to_keep <- integrated_sce$sample %in% library_ids
+rms_sce <- integrated_sce[,samples_to_keep]
+```
```

Comment:
It looks like this notebook got killed in CI for memory usage. We probably want to get rid of the integrated object as soon as we subset it here. If that doesn't work, we may need to think about other solutions.